Merge BZ and GH into one place

mfalesni opened the pull request:
* BZ and GH are now specified in the meta marker
* some of the unused BZ functionality removed
* GH functionality extended (same level as BZ currently)

{{pytest: -v cfme/tests/automate/test_buttons.py cfme/tests/automate/test_customization_paginator.py cfme/tests/automate/test_domain_priority.py cfme/tests/automate/test_namespace.py cfme/tests/automate/test_provisioning_dialogs.py cfme/tests/cloud/test_cloud_timelines.py cfme/tests/cloud/test_instance_power_control.py cfme/tests/cloud/test_provisioning.py cfme/tests/configure/test_access_control.py cfme/tests/configure/test_db_backup_schedule.py cfme/tests/configure/test_docs.py cfme/tests/configure/test_log_depot_operation.py cfme/tests/configure/test_register_appliance.py cfme/tests/control/test_actions.py cfme/tests/control/test_bugs.py cfme/tests/control/test_import_policies.py cfme/tests/infrastructure/test_advanced_search_host.py cfme/tests/infrastructure/test_advanced_search_providers.py cfme/tests/infrastructure/test_advanced_search_vms.py cfme/tests/infrastructure/test_customization_template.py cfme/tests/infrastructure/test_datastore_analysis.py cfme/tests/infrastructure/test_genealogy.py cfme/tests/infrastructure/test_host_analysis.py cfme/tests/infrastructure/test_vm_clone.py cfme/tests/infrastructure/test_vm_migrate.py cfme/tests/intelligence/reports/test_menus.py cfme/tests/intelligence/test_chargeback.py cfme/tests/intelligence/test_dashboard.py cfme/tests/perf/ui/test_ui_cloud.py cfme/tests/perf/ui/test_ui_control.py cfme/tests/perf/ui/test_ui_infrastructure.py cfme/tests/perf/ui/test_ui_intelligence.py cfme/tests/perf/ui/test_ui_services.py cfme/tests/services/test_catalog.py cfme/tests/services/test_catalog_item.py cfme/tests/services/test_cloud_service_catalogs.py cfme/tests/services/test_generic_service_catalogs.py cfme/tests/services/test_iso_service_catalogs.py cfme/tests/services/test_myservice.py cfme/tests/services/test_pxe_service_catalogs.py cfme/tests/services/test_service_catalogs.py cfme/tests/test_appliance.py cfme/tests/test_soap.py --use-provider vsphere55 --long-running --capture=no}}

dajoRH commented:
Lint report for commit 94227b3c60167f99ef12ebd19dab9ac96d4a61da:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/configure/settings.py:9:26: E201 whitespace after '('
./cfme/configure/settings.py:10:27: E202 whitespace before ')'
```
*CFME QE Bot*

psav commented on cfme/tests/test_appliance.py:
```diff
@@ -139,7 +139,7 @@ def test_db_connection(db):
 
 
-@pytest.mark.bugzilla(1121202)
+@pytest.mark.meta(blockers=["BZ#1121202"])
```
Why is this one different from all the others?

mfalesni replied:
Because I made this one before I started to mass-change all of them so ... the functionality is the same though. I'll change it if you want.

dajoRH commented:
Lint report for commit 27730e3eb5ab136e59e3b3cb583d9170935bf9cd:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/configure/settings.py:9:26: E201 whitespace after '('
./cfme/configure/settings.py:10:27: E202 whitespace before ')'
```
*CFME QE Bot*

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

dajoRH commented:
Lint report for commit 40cf5b7c4d0e06267a87c1636aebc2bdeb173630:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./utils/trackerbot.py:99:5: E125 continuation line with same indent as next logical line
```
*CFME QE Bot*

dajoRH commented:
Lint report for commit fcb00294d9c06d8287f321d510431fab9c6c73fe:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

seandst commented on metaplugins/blockers.py:
```diff
@@ -0,0 +1,135 @@
+            GH("owner/repo:issue"), # Otherwise you need to use this syntax
+            123456,                 # Will resolve to BZ
+            "BZ#123456",            # Will resolve to BZ too
+            "Issue #123",           # Will resolve to GH (you need default repo set for this)
```
I'm thinking it'd be good to simplify around a limited set, and probably always identify what tracker to use and use string literals.

e.g. `"BZ#123456"`, `"GH#123"`, `"GH#owner/repo:issue"`. The might(?) simplify things below a little bit, since it allows for getting rid of the regular expressions (`tracker, bugid = blocker.split("#")`)

mfalesni commented:
@seandst I have changed that. I still keep some of the shorter writing styles but I added the generic one as you mentioned. Plus some small tweaks of course.

seandst commented:
> I have changed that. I still keep some of the shorter writing styles but I added the generic one as you mentioned. Plus some small tweaks of course.

I think the generic ones were already there. That said, I'm really interested in only being able to identify a bug one way, not 3 or 4 different-looking but functionally identical ways. In addition to that, I'd rather not have to bring in special classes just to mark a test as being affected by a bug. So having the "tracker#bug_ident" interface is pretty much the only thing I want to see be supported.

mfalesni commented:
@seandst Not even the just number for bugzilla?

seandst commented:
> @seandst Not even the just number for bugzilla?

Correct, not even that. Although if you put in something that takes ints and appends the "BZ#" before processing, that'll be good.

This tool is very likely to be one that we spin off into its own package, so defaulting to BZ should be configurable. I'm trying to find the happy medium between getting what we want for ourselves merged now without making it harder in the future to split things off.

mfalesni commented:
@seandst Ok, I extracted the entire Blockers logic in the utils/blockers.py module, I just left the bugzilla shortcut on line 88 in metaplugins/blockers.py because I don't know about better place to put it right now without making significant fuss. You can see the latest commit unsquashed.

seandst commented:
I think this is fine for merging, but I'll leave it to you (@mfalesni) to decide if you want to squash it down to one commit or not.

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

dajoRH commented:
Lint report for commit ad679197a5454da5254e912766944f3524ddd528:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

dajoRH commented:
Lint report for commit 5a7fccb1321956e0120a26b4249ba8807f386115:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

dajoRH commented:
Lint report for commit 18bdb50d48a0de57beb76cbd211a6fa097fecbd3:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

dajoRH commented:
Requirements have changed. @seandst , @psav ?
*CFME QE Bot*

dajoRH commented:
Lint report for commit a50d936772bb4278d169b652dd64d08059fe657c:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

mfalesni commented:
Rebased, another round of tests then :cry: I would merge it because apart from one thing that I've fixed it worked properly.

dajoRH commented:
Lint report for commit 42c8e717d1d389d0f538f24024a4a420f976eab9:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

dajohnso commented:
@mfalesni , it looks like we are losing the bug fixture to pass to tests and get them to still run and only skip a failing assertion, etc.  That is still a requirement and something that I want to get more people using.

dajoRH commented:
Lint report for commit ae60783bec074d870f8a661f351876dc205d2d7c:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

dajoRH commented:
Lint report for commit 16f05a420c6b3b3ac0612d0d55025c397a07124b:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

mfalesni commented:
I also added additional BZ mark for the log depot collection (1184465).
